core/modules/taxonomy/taxonomy.module

```diff
@@ -1852,6 +1857,32 @@ function taxonomy_field_widget_form(&$form, &$form_state, $field, $instance, $la
   return $element;
 }
 
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function taxonomy_field_widget_form_alter(&$element, &$form_state, $context) {
+  if ($context['field']['type'] == 'taxonomy_term_reference' && in_array($element['#type'], array(
```

Comment:
This if can be simplified by removing the check for element type (see my comment above).

Reply:
Agreed.